app/frontend/src/service/storage.ts

```diff
@@ -1,149 +1,314 @@
+                    storedAnswers.push(a);
+                }
+                result.Data.LastEdited = Date.now();
+                if (storage.objectStore_name === "chat") {
+                    result.Options.system = system_message;
+                    result.Options.maxTokens = max_tokens;
+                    result.Options.temperature = temperature;
+                }
+                data = result;
             } else {
-                request = chat.put({ Answers: [a], id: ID });
+                let name: string = "";
+                let new_idcounter = id_counter;
+                if (language != undefined && temperature != undefined && system_message != undefined && max_tokens != undefined) {
+                    name = await (await getChatName(a, language, temperature, system_message, max_tokens)).content;
+                    name = name.replaceAll('"', "").replaceAll(".", "");
+                }
+                if (storage.objectStore_name === "chat") {
+                    new_idcounter = new_idcounter + 1;
+                    setIdCounter(new_idcounter);
+                    data = {
+                        Data: { Answers: [a], Name: name, LastEdited: Date.now() },
+                        id: new_idcounter,
+                        Options: { favorite: false, system: system_message, maxTokens: max_tokens, temperature: temperature }
+                    };
+                } else {
+                    data = {
+                        Data: { Answers: [a], Name: name, LastEdited: Date.now() },
+                        id: new_idcounter
+                    };
+                }
+                setCurrentId(new_idcounter);
+                dataID = new_idcounter;
             }
-            request.onerror = function () {
-                console.log("Error", request.error);
-            };
+            let chat = openRequest.result.transaction(storage.objectStore_name, "readwrite").objectStore(storage.objectStore_name);
+            let request = chat.put(data);
+            request.onerror = () => onError(request);
+            data["id"] = CURRENT_CHAT_IN_DB;
+            data["refID"] = dataID;
+            request = chat.put(data);
+            request.onerror = () => onError(request);
         };
     };
 }
 
-export async function getStartDataFromDB(storage: indexedDBStorage): Promise<any> {
-    let openRequest = indexedDB.open(storage.db_name, VERSION);
-    let db;
-    openRequest.onupgradeneeded = function () {
-        db = openRequest.result;
-        if (!db.objectStoreNames.contains(storage.objectStore_name)) {
-            db.createObjectStore(storage.objectStore_name, { keyPath: "id" });
-        }
+export async function getChatName(answers: any, language: string, temperature: number, system_message: string, max_tokens: number) {
+    const history: ChatTurn[] = [{ user: answers[0], bot: answers[1].answer }];
+    const request: ChatRequest = {
+        history: [
+            ...history,
```

Comment:
Hier würde ich wahrscheinlich nicht die ganze History hinschicken, sondern nur maximal die ersten 3 Konversationen (Frage/Antwort) Paare. Nicht das wir hier in ein Tokenlimit laufen.

Reply:
Die Methode wird nur nach der ersten Nachricht aufgerufen, somit wird immer nur ein Frage/Antwort Paar mitgeschickt.